src/css.js

```diff
@@ -133,21 +130,33 @@ function getWidthOrHeight( elem, dimension, extra ) {
 		val = "auto";
 	}
 
-	// Check for style in case a browser which returns unreliable values
-	// for getComputedStyle silently falls back to the reliable elem.style
-	valueIsBorderBox = valueIsBorderBox &&
-		( support.boxSizingReliable() || val === elem.style[ dimension ] );
+	isBorderBox = jQuery.css( elem, "boxSizing", false, styles ) === "border-box";
+	valueIsBorderBox = isBorderBox && support.boxSizingReliable();
+	offsetProp = "offset" + dimension[ 0 ].toUpperCase() + dimension.slice( 1 );
 
 	// Fall back to offsetWidth/offsetHeight when value is "auto"
 	// This happens for inline elements with no explicit setting (gh-3571)
 	// Support: Android <=4.1 - 4.3 only
 	// Also use offsetWidth/offsetHeight for misreported inline dimensions (gh-3602)
-	if ( val === "auto" ||
-		!parseFloat( val ) && jQuery.css( elem, "display", false, styles ) === "inline" ) {
-
-		val = elem[ "offset" + dimension[ 0 ].toUpperCase() + dimension.slice( 1 ) ];
+	// Support: IE 9-11 only
+	// Also use offsetWidth/offsetHeight for when box sizing is unreliable
+	// Do not use offset for elements that don't have it (e.g. SVG)
+	if ( typeof elem[ offsetProp ] !== "undefined" &&
+		( val === "auto" || ( isBorderBox && !valueIsBorderBox ) ||
+			!parseFloat( val ) && jQuery.css( elem, "display", false, styles ) === "inline" ) ) {
+
+		offsetVal = elem[ offsetProp ];
+
+		// Support: IE 9-11 only
+		// offsetWidth/offsetHeight is zero for disconnect elements
+		// and children of hidden elements
+		if ( offsetVal || !parseFloat( val ) ) {
+			val = offsetVal;
+		}
```

Comment:
Big size benefit from dropping `offsetVal` entirely and doing this like
```js
val = elem[ offsetProp ] || parseFloat( val ) || 0;
```
Plus one more from also dropping `offsetProp`:
```js
val = elem[ "offset" + dimension[ 0 ].toUpperCase() + dimension.slice( 1 ) ] ||
	parseFloat( val ) || 0;
```

Reply:
I can do the first, but still need the second to check for existence of the property.

Reply:
You can check for property existence by repeating `elem[ "offset" + dimension[ 0 ].toUpperCase() + dimension.slice( 1 ) ]`, which works in our favor here.

Reply:
I don't think we should do that, even if it's smaller.

Reply:
Although it may be moot now. Now that it falls back to val, `undefined` is fine.

Reply:
Oh wait, that's only if we want `valueIsBorderBox` to be `true` for SVG. From my testing, it seems like it should be `false`. I'll wait for your reply on the jsbin.